diff --git a/externs/google_Maps/target/downloads/google_maps_api_v3_11.js b/externs/google_Maps/target/downloads/google_maps_api_v3_11.js
index ac91685..5dbb628 100644
--- a/externs/google_Maps/target/downloads/google_maps_api_v3_11.js
+++ b/externs/google_Maps/target/downloads/google_maps_api_v3_11.js
@@ -20,6 +20,11 @@
  * @externs
  */
 
+
+/** @const */
+var google = {};
+
+/** @const */
 google.maps = {};
 
 /**
@@ -4181,7 +4186,7 @@ google.maps.ZoomControlStyle = {
   SMALL: 3
 };
 
-// Namespace
+/** @const */
 google.maps.adsense = {};
 
 /**
@@ -4386,7 +4391,7 @@ google.maps.adsense.AdUnitOptions.prototype.titleColor;
  */
 google.maps.adsense.AdUnitOptions.prototype.urlColor;
 
-// Namespace
+/** @const */
 google.maps.drawing = {};
 
 /**
@@ -4517,7 +4522,7 @@ google.maps.drawing.OverlayType = {
   RECTANGLE: ''
 };
 
-// Namespace
+/** @const */
 google.maps.event = {};
 
 /**
@@ -4581,10 +4586,10 @@ google.maps.event.removeListener = function(listener) {};
  */
 google.maps.event.trigger = function(instance, eventName, var_args) {};
 
-// Namespace
+/** @const */
 google.maps.geometry = {};
 
-// Namespace
+/** @const */
 google.maps.geometry.encoding = {};
 
 /**
@@ -4599,7 +4604,7 @@ google.maps.geometry.encoding.decodePath = function(encodedPath) {};
  */
 google.maps.geometry.encoding.encodePath = function(path) {};
 
-// Namespace
+/** @const */
 google.maps.geometry.poly = {};
 
 /**
@@ -4617,7 +4622,7 @@ google.maps.geometry.poly.containsLocation = function(point, polygon) {};
  */
 google.maps.geometry.poly.isLocationOnEdge = function(point, poly, opt_tolerance) {};
 
-// Namespace
+/** @const */
 google.maps.geometry.spherical = {};
 
 /**
@@ -4682,7 +4687,7 @@ google.maps.geometry.spherical.computeSignedArea = function(loop, opt_radius) {}
  */
 google.maps.geometry.spherical.interpolate = function(from, to, fraction) {};
 
-// Namespace
+/** @const */
 google.maps.panoramio = {};
 
 /**
@@ -4819,7 +4824,7 @@ google.maps.panoramio.PanoramioMouseEvent.prototype.latLng;
  */
 google.maps.panoramio.PanoramioMouseEvent.prototype.pixelOffset;
 
-// Namespace
+/** @const */
 google.maps.places = {};
 
 /**
@@ -5486,7 +5491,7 @@ google.maps.places.TextSearchRequest.prototype.radius;
  */
 google.maps.places.TextSearchRequest.prototype.types;
 
-// Namespace
+/** @const */
 google.maps.visualization = {};
 
 /**
@@ -5910,7 +5915,7 @@ google.maps.visualization.WeightedLocation.prototype.location;
  */
 google.maps.visualization.WeightedLocation.prototype.weight;
 
-// Namespace
+/** @const */
 google.maps.weather = {};
 
 /**
